### ISSUE TITLE
deno 0.5.0 - http server crash with incoming https request

### ISSUE THREAD

**Issue by JoakimCh** [denoland/deno#2346]
The http server will crash if it has an incoming https request, with this error:
> Uncaught ProtocolError: malformed MIME header line

In http lib's `readRequest` this line causes the error:
`[req.headers, err] = await tp.readMIMEHeader()`

It's trying to read the https request, but since it's encrypted it fails at doing this. But there should be a way to detect such requests and just deny them without crashing the server.

**Comment by zekth** [comment]
@ry as HTTPS is not implemented yet a solution would be to do a redirect on http protocol? Or return an error 503?

**Comment by ry** [comment]
@zekth return 400
```
$ python -m SimpleHTTPServer 8000
Serving HTTP on 0.0.0.0 port 8000 ...
127.0.0.1 - - [13/May/2019 18:12:23] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [13/May/2019 18:12:29] code 400, message Bad request syntax ('\x16\x03\x01\x00\xd7\x01\x00\x00\xd3\x03\x03\x92\xdb\xf9\xb5m\x9c\x1b\x1aH\xc9\x9f\xb2~\x92\xf1x.\x9f\xfb\xe3\x1b\xe9\xba+\xff)db\x12\x81\xe3\x97\x00\x00T\xcc\xa9\xcc\xa8\xcc\xaa\xc00\xc0,\xc0(\xc0$\xc0\x14\xc0')
127.0.0.1 - - [13/May/2019 18:12:29] "�����m�Hɟ�~��x.����+�)db��T̨̩̪�0�,�(�$��" 400 -
```
in another terminal run:
```
$ curl https://localhost:8000/
curl: (35) error:1400410B:SSL routines:CONNECT_CR_SRVR_HELLO:wrong version number
```

**Comment by bartlomieju** [comment]
Resolved in denoland/deno_std#415?

**Comment by zekth** [comment]
> Resolved in [denoland/deno_std#415](https://github.com/denoland/deno_std/pull/415)?

not related. This PR solve the problem of handling the end of the mime header but not encrypted mime. So it will still try to read the mime header but will fail.
